File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
from collections import deque

def solution(s):
    x=len(s)
    answer=0
    arr=deque(s)
    for i in range(x):
        stack=[]
        for j in arr:
            if len(stack)==0:
                stack.append(j)
            else:
                if stack[-1]=='[' and j==']':
                    stack.pop()
                elif stack[-1]=='(' and j==')':
                    stack.pop()
                elif stack[-1]=='{' and j=='}':
                    stack.pop()
                else:
                    stack.append(j)
        if stack==[]:
            answer+=1
        arr.append(arr.popleft())

                    
    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int x = s.length();
        int answer = 0;
        ArrayDeque<Character> arr = new ArrayDeque<>();
        char[] ch = s.toCharArray();
        for(char i : ch){
            arr.add(i);
        }
        // System.out.println(arr);
        
        for(char i : ch){
            ArrayDeque<Character> stack = new ArrayDeque<>();
            for(char j : arr){
                if(stack.size()==0){
                    stack.add(j);
                }else{
                    if(stack.peekLast()=='[' && j==']'){
                        stack.pollLast();
                    }else if(stack.peekLast()=='(' && j==')'){
                        stack.pollLast();
                    }else if(stack.peekLast()=='{' && j=='}'){
                        stack.pollLast();
                    }else{
                        stack.add(j);
                    }
                }
                
            }if(stack.size()==0){
                answer+=1;
            }
            arr.add(arr.poll());
        }
        
        
        
        return answer;
    }
}